src/ui/public/chrome/api/nav.js

```diff
@@ -6,6 +6,15 @@ export default function (chrome, internals) {
     return internals.nav;
   };
 
+  chrome.getNavLinkById = (id) => {
+    const navLink = internals.nav.find(link => link.id === id);
+    if (navLink) {
+      return navLink;
+    } else {
+      throw new Error(`Nav link for id = ${id} not found`);
```

Comment:
In the spirit of [Return early from functions](https://github.com/epixa/kibana/blob/styleguide/style_guides/js_style_guide.md#return-early-from-functions), I recommend doing this assertion up front and then returning outside of the conditional:

```js
const navLink = internals.nav.find(link => link.id === id);
if (!navLink) {
  throw new Error(`Nav link for id = ${id} not found`);
}
return navLink;
```